src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1257,53 +1242,41 @@ public StringBuilder Append(ReadOnlySpan<char> value)
 
         #region AppendJoin
 
-        public unsafe StringBuilder AppendJoin(string? separator, params object?[] values)
+        public StringBuilder AppendJoin(string? separator, params object?[] values)
         {
             separator ??= string.Empty;
```

Comment:
This should no longer be necessary.  `separator.AsSpan()` will return an empty span if `separator` is null.